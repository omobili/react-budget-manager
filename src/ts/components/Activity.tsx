import * as React from "react";
import {dataService} from "../services/dataService";

export interface ActivityProps {
    id?: number;
    label: string;
    amount: number;
}

export interface ActivityState {

}

export class Activity extends React.Component<ActivityProps, ActivityState> {
    constructor(props: ActivityProps) {
        super(props);

        this.handleDelete = this.handleDelete.bind(this);
    }

    handleDelete(e: any) {
        e.preventDefault();

        if (this.props.id) {
            dataService.removeActivity(this.props.id);
        }
    }

    render() {
        let activityClassName = function (amount: number) {
            return `rbm-activity${(amount > 0) ? ' rbm-activity--plus': ' rbm-activity--minus'}`;
        };

        return (
            <div className={activityClassName(this.props.amount)}>
                <div className="row">
                    <span className="col rbm-activity__label">{this.props.label}</span>
                    <span className="col-auto rbm-activity__amount">{this.props.amount} â‚¬</span>
                    <span className="col-auto rbm-activity__delete">
                        <button onClick={this.handleDelete} className="btn btn-danger btn-sm">Supprimer</button>
                    </span>
                </div>
            </div>
        )
    }
}