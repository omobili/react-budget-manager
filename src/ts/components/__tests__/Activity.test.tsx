import * as React from "react";
import {Activity} from "../Activity";
import * as renderer from "react-test-renderer";

describe('Activity', () => {
    describe('Snapshots', () => {
        test('should match snapshot for a negative amount', () => {
            let activity = renderer.create(
                <Activity label={"depense"} amount={-12.45} />,
            );

            let tree = activity.toJSON();
            expect(tree).toMatchSnapshot();
        });

        test('should match snapshot for a positive amount', () => {
            let activity = renderer.create(
                <Activity label={"recette"} amount={12.45} />,
            );

            let tree = activity.toJSON();
            expect(tree).toMatchSnapshot();
        });

        test('should match snapshot for given id (0)', () => {
            let activity = renderer.create(
                <Activity label={"depense"} amount={-12.45} id={0} />,
            );

            let tree = activity.toJSON();
            expect(tree).toMatchSnapshot();
        });

        test('should match snapshot for given id (> 0)', () => {
            let activity = renderer.create(
                <Activity label={"depense"} amount={-12.45} id={123} />,
            );

            let tree = activity.toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
});

