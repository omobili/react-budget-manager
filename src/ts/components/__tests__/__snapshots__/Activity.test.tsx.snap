// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Activity Snapshots should match snapshot for a negative amount 1`] = `
<div
  className="rbm-activity rbm-activity--minus"
>
  <div
    className="row"
  >
    <span
      className="col rbm-activity__label"
    >
      depense
    </span>
    <span
      className="col-auto rbm-activity__amount"
    >
      -12.45
       €
    </span>
  </div>
</div>
`;

exports[`Activity Snapshots should match snapshot for a positive amount 1`] = `
<div
  className="rbm-activity rbm-activity--plus"
>
  <div
    className="row"
  >
    <span
      className="col rbm-activity__label"
    >
      recette
    </span>
    <span
      className="col-auto rbm-activity__amount"
    >
      12.45
       €
    </span>
  </div>
</div>
`;

exports[`Activity Snapshots should match snapshot for given id (> 0) 1`] = `
<div
  className="rbm-activity rbm-activity--minus"
>
  <div
    className="row"
  >
    <span
      className="col rbm-activity__label"
    >
      depense
    </span>
    <span
      className="col-auto rbm-activity__amount"
    >
      -12.45
       €
    </span>
    <span
      className="col-auto rbm-activity__delete"
    >
      <button
        className="btn btn-danger btn-sm"
        onClick={[Function]}
      >
        Supprimer
      </button>
    </span>
  </div>
</div>
`;

exports[`Activity Snapshots should match snapshot for given id (0) 1`] = `
<div
  className="rbm-activity rbm-activity--minus"
>
  <div
    className="row"
  >
    <span
      className="col rbm-activity__label"
    >
      depense
    </span>
    <span
      className="col-auto rbm-activity__amount"
    >
      -12.45
       €
    </span>
    <span
      className="col-auto rbm-activity__delete"
    >
      <button
        className="btn btn-danger btn-sm"
        onClick={[Function]}
      >
        Supprimer
      </button>
    </span>
  </div>
</div>
`;
